### PR TITLE
Run preflights inline

### PR DESCRIPTION
**Pull request by marccampbell**
Removing the deploysocket to run preflights and just using the kotsadm vendor'ed version of preflight.

This introduces a regression that must be solved before we can release the next version: `kots upload` will not trigger preflights because it's uploading to the TS API. That's code that needs to change in KOTS, so I'll move there next.

### CODE REVIEW

File: pkg/preflight/preflight.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+			return errors.Wrap(err, "failed to load rendered preflight")
+		}
+
+		go func() {
+			if err := execute(appID, sequence, p); err != nil {
+				logger.Error(err)
+				return
+			}
+
+			logger.Debug("preflight checks completed")
```
Comment (laverya) [inline pkg/preflight/preflight.go]: should we have a 'starting preflight checks' log too?

File: pkg/kotsutil/yaml.go [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+import (
+	"github.com/pkg/errors"
+	"github.com/replicatedhq/kots/pkg/util"
+	yaml "github.com/replicatedhq/yaml/v3"
+)
+
+// FixUpYAML is a general purpose function that will ensure that YAML is copmatible with KOTS
+// This ensures that lines aren't wrapped at 80 chars which breaks template functions
+func FixUpYAML(inputContent []byte) ([]byte, error) {
```
Comment (laverya) [inline pkg/kotsutil/yaml.go]: nice

File: pkg/preflight/execute.go [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+	"github.com/replicatedhq/kotsadm/pkg/logger"
+	"github.com/replicatedhq/kotsadm/pkg/persistence"
+	troubleshootv1beta1 "github.com/replicatedhq/troubleshoot/pkg/apis/troubleshoot/v1beta1"
+	troubleshootpreflight "github.com/replicatedhq/troubleshoot/pkg/preflight"
+	"go.uber.org/zap"
+	"k8s.io/client-go/rest"
+)
+
+// execute will execute the preflights using spec in renderedPreflights.
+// This spec should be rendered, no template functions remaining
```
Comment (laverya) [inline pkg/preflight/execute.go]: this comment feels wrong, since the func is passed an object

File: pkg/version/version.go [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+	configSpec, err := kotsKinds.Marshal("kots.io", "v1beta1", "Config")
+	if err != nil {
+		return int64(0), errors.Wrap(err, "failed to marshal config spec")
+	}
+	configValuesSpec, err := kotsKinds.Marshal("kots.io", "v1beta1", "ConfigValues")
+	if err != nil {
+		return int64(0), errors.Wrap(err, "failed to marshal configvalues spec")
+	}
+
+	// TODO: solve
```
Comment (laverya) [inline pkg/version/version.go]: this TODO is very vague

File: pkg/version/version.go [hunk 5]
```diff
@@ -0,0 +1,307 @@
...
+	registrySettings, err := getRegistrySettingsForApp(appID)
+	if err != nil {
+		return int64(0), errors.Wrap(err, "failed to get registry settings")
+	}
+
+	preflightBytes, err := render.RenderFile(kotsKinds, registrySettings, []byte(preflightSpec))
+	if err != nil {
+		return 0, errors.Wrap(err, "failed to template preflight spec")
+	}
+	templatedPreflight := string(preflightBytes)
```
Comment (laverya) [inline pkg/version/version.go]: is this still required, with the new inline preflight caller?

File: pkg/version/version.go (hunk 4 above)
Comment (marccampbell) [reply pkg/version/version.go]: ha! i fixed the code but forgot to remove the todo!

File: pkg/version/version.go (hunk 5 above)
Comment (marccampbell) [reply pkg/version/version.go]: hmmm, good question..

Comment (marccampbell) [reply pkg/version/version.go]: actually why do we even store them in the database?  i'm going to start storing the unrendered version, but not change the schema to drop that column for now.

File: pkg/preflight/preflight.go (hunk 1 above)
Comment (marccampbell) [reply pkg/preflight/preflight.go]: added

File: pkg/preflight/execute.go (hunk 3 above)
Comment (marccampbell) [reply pkg/preflight/execute.go]: It's correct.  the var name is wrong, but it's a rendered and unmarshalled object, not a string.

updating var name in the comment

File: pkg/version/version.go (hunk 5 above)
Comment (laverya) [reply pkg/version/version.go]: because the TS read the preflight specs from the database

File: api/src/sockets/kots/DeploySocket.ts [hunk 6]
```diff
@@ -260,17 +260,6 @@ export class KotsDeploySocketService {
         if (app.restoreInProgressName) {
           continue;
         }
-        const pendingPreflightParams = await this.preflightStore.getPendingPreflightParams(true);
```
Comment (divolgin) [inline api/src/sockets/kots/DeploySocket.ts]: is this now gone? does it need a TODO?

Comment (divolgin) [reply api/src/sockets/kots/DeploySocket.ts]: nm, i think i found it

File: pkg/handlers/config.go [hunk 7]
```diff
@@ -229,47 +235,46 @@ func updateAppConfig(updateApp *app.App, seq int, req UpdateAppConfigRequest, is
...
-		}
-
-		if err := app.CreateAppVersionArchive(updateApp.ID, int64(seq), archiveDir); err != nil {
+		if err := version.CreateAppVersionArchive(updateApp.ID, int64(sequence), archiveDir); err != nil {
 			updateAppConfigResponse.Error = "failed to create app version archive"
 			return updateAppConfigResponse, err
 		}
 	}
 
+	if err := preflight.Run(updateApp.ID, int64(sequence), archiveDir); err != nil {
```
Comment (divolgin) [inline pkg/handlers/config.go]: i guess we'll see if this is better than running them in a loop.  I think this breaks the "retry on missing RBAC" logic.

File: api/src/sockets/kots/DeploySocket.ts [hunk 8]
```diff
@@ -260,17 +260,6 @@ export class KotsDeploySocketService {
...
-        for (const param of pendingPreflightParams) {
-          if (clusterSocketHistory.sentPreflightUrls[param.url] !== param.ignorePermissions) {
```
Comment (divolgin) [inline api/src/sockets/kots/DeploySocket.ts]: this is where the "Run anyway" button is honored.

**Pull request merged**